### DISCUSSION TITLE
What could be the possible reason for sahi performance is lower than yolov5  standard output

### DISCUSSION THREAD

**Discussion by imadgohar** [obss/sahi#683] [Q&A] [answered]
I trained a custom dataset with yolov5, and on test data I am getting 90% mAP50 but when I apply sahi on the same test set the results are very poor.
I tried sahi with different window sizes starting from 256 to 1500 range.

On larger window size the max result with sahi reached to mAP50 to around 50%.

What should I am doing wrong? Why the results are lower than the original yolov5 results.

I am using aerial images containing both large and small objects.

Any suggestions?

**Comment by fcakyon** [comment, marked as answer]
You should use nms as postprocessing metric and threshold as 0.01 to get high ap50 with sahi.

sahi uses yolov5 pypi package hublike inference api which is slower than yolov5 detect/val scripts.

You can also see the inference parameters that we used for the ICIP paper here: https://github.com/fcakyon/small-object-detection-benchmark/blob/main/eval_tools/predict_evaluate_analyse.py

Feel free to ask more questions if you have.

**Reply by imadgohar** [reply]
I am using NMS, let's change the threshold value. thanks

**Reply by fcakyon** [reply]
Also, try not to use smaller slice sizes.

**Reply by imadgohar** [reply]
Alright. Many thanks for good suggestions.
